Zend/zend_strtod.c

```diff
@@ -2606,7 +2606,7 @@ zend_strtod
 		}
 	s0 = s;
 	y = z = 0;
-	for(nd = nf = 0; (c = *s) >= '0' && c <= '9'; nd++, s++)
+	for(nd = nf = 0; (c = *s) >= '0' && c <= '9' && nd < INT_MAX; nd++, s++)
```

Comment:
So you abort the loop on overflow but don't signal an error. Does this result in a wrong resulting number?

Reply:
You can find _derivations_ of the original code here and there, even netbsd uses in its own libc, but no mention of such fix.